tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -170,13 +171,28 @@ pub mod pallet {
 			);
 			ensure!(identity != who, Error::<T>::LinkPrimeIdentityDisallowed);
 
-			ensure!(
-				identity.matches_web3networks(web3networks.as_ref()),
-				Error::<T>::WrongWeb3NetworkTypes
-			);
+			let mut adjusted_web3networks = web3networks;
+
+			// To reduce the encoded call size, the client has a workaround on `staging` branch to
+			// pass in empty `web3networks`, we'll need to restore it to full network types
+			if identity.is_web3() && adjusted_web3networks.is_empty() {
+				adjusted_web3networks = match identity {
+					Identity::Substrate(..) => all_substrate_web3networks(),
+					Identity::Evm(..) => all_evm_web3networks(),
+					Identity::Bitcoin(..) => all_bitcoin_web3networks(),
+					_ => Vec::new(),
```

Comment:
shouldn't it throw here and panic?

Reply:
Normally we don't panic inside pallet, it's better to return an error here - but since it's on staging, I didn't even bother adding another error type (which requires refreshment of sidechain metadata) as I know it's unreachable.